sys/net/gnrc/network_layer/icmpv6/error/gnrc_icmpv6_error.c

```diff
@@ -234,24 +234,36 @@ static void _send(gnrc_pktsnip_t *pkt, const gnrc_pktsnip_t *orig_pkt,
     }
 }
 
-static gnrc_pktsnip_t *_check_ipv6_hdr(const gnrc_pktsnip_t *orig_pkt)
+static gnrc_pktsnip_t *_check_ipv6_hdr(const gnrc_pktsnip_t *orig_pkt,
+                                       uint8_t type, uint8_t code)
 {
     /* discarding const qualifier is safe here */
     gnrc_pktsnip_t *ipv6 = gnrc_pktsnip_search_type((gnrc_pktsnip_t *)orig_pkt,
                                                     GNRC_NETTYPE_IPV6);
     assert(ipv6 != NULL);
     const ipv6_hdr_t *ipv6_hdr = ipv6->data;
 
+    /* see https://tools.ietf.org/html/rfc4443#section-2.4 (e.6) */
     if (ipv6_addr_is_unspecified(&ipv6_hdr->src) ||
-        ipv6_addr_is_multicast(&ipv6_hdr->dst)) {
+        ipv6_addr_is_multicast(&ipv6_hdr->src)) {
         ipv6 = NULL;
     }
+    /* ipv6_hdr->dst may only be multicast for Packet Too Big Messages
+     * and Parameter Problem Messages with code 2, see
+     * https://tools.ietf.org/html/rfc4443#section-2.4 (e.3) */
+    else if (ipv6_addr_is_multicast(&ipv6_hdr->dst)) {
+        if ((type != ICMPV6_PKT_TOO_BIG) ||
+            ((type == ICMPV6_PARAM_PROB) &&
+             (code != ICMPV6_ERROR_PARAM_PROB_OPT))) {
+            ipv6 = NULL;
+        }
```

Comment:
This condition is always true if `type != ICMPV6_PKT_TOO_BIG`.

Reply:
Yepp, that's the point. From [RFC 4443, section 2.4](https://tools.ietf.org/html/rfc4443#section-2.4) (`ipv6 == NULL` will result in the sending functions being aborted, see code below):

>     (e) An ICMPv6 error message MUST NOT be originated as a result of
>         receiving the following:
>          […]
>          (e.3) A packet destined to an IPv6 multicast address.  (There are
>                two exceptions to this rule: (1) the Packet Too Big Message
>                (Section 3.2) to allow Path MTU discovery to work for IPv6
>                multicast, and (2) the Parameter Problem Message, Code 2
>                (Section 3.4) reporting an unrecognized IPv6 option (see
>                Section 4.2 of [IPv6]) that has the Option Type highest-
>                order two bits set to 10).

Reply:
But it's also true if `type == ICMPV6_PARAM_PROB && code == ICMPV6_ERROR_PARAM_PROB_OPT`

Reply:
`type != ICMPV6_PKT_TOO_BIG` already evaluates to `true`, so it's irrelevant what's in the second `or` case.

Reply:
You are right. I did not find time to address this yet though. Will fix ASAP.

Reply:
Should be fixed now, but I did not retest yet.